daemon/cmd/ipam.go

```diff
@@ -240,6 +241,20 @@ func (d *Daemon) allocateDatapathIPs(family types.NodeAddressingFamily) (routerI
 		node.SetRouterInfo(routingInfo)
 	}
 
+	cidrs := make([]*net.IPNet, 0, len(result.CIDRs))
+	for _, k := range result.CIDRs {
+		var s net.IPNet
+		ip, mask, _ := net.ParseCIDR(k)
+		s.IP = ip
+		s.Mask = mask.Mask
+		cidrs = append(cidrs, &s)
+	}
+	resultcidr, _ := iputil.CoalesceCIDRs(cidrs)
+	newresult := make([]string, len(resultcidr))
+	for i, k := range resultcidr {
+		newresult[i] = k.String()
+	}
+	result.CIDRs = newresult
```

Comment:
This seems identical to the code block below? Can you extract it into a separate function? Thanks

Reply:
Moved to coalesce functionality to a function.